idanalysis/eqparams.py

```diff
@@ -0,0 +1,1194 @@
+"""IDs analysis."""
+
+import mathphys
+import matplotlib.pyplot as _plt
+import numpy as _np
+import pyaccel
+import pymodels
+from fieldmaptrack import Beam
+from scipy.integrate import cumtrapz
+from scipy.optimize import minimize
+
+
+class InsertionParams:
+    """Class to specify insertion parameters.
+
+     All equations used here are from the book: Clarke, James A.,
+    'The Science and Technology of Undulators and Wigglers'.
+    Chapter 10: 'The Effect of Insertion Devices on the Electron Beam'.
+    """
+
+    def __init__(self):
+        """Class constructor."""
+        self._field_profile = None
+        self._fam_name = None
+        self._bx_peak = None
+        self._by_peak = None
+        self._kx = None
+        self._ky = None
+        self._period = None
+        self._length = None
+        self._nr_periods = None
+        self._straight_section = None
+        self._etax = None
+        self._etapx = None
+        self._etay = None
+        self._etapy = None
+        self._i1x = None
+        self._i2x = None
+        self._i3x = None
+        self._i4x = None
+        self._i5x = None
+        self._i1y = None
+        self._i2y = None
+        self._i3y = None
+        self._i4y = None
+        self._i5y = None
+        self._u0 = None
+        self.brho = Beam(3).brho
+        self._e = mathphys.constants.elementary_charge
```

Comment:
I made them module constants, however, I decided to keep them in lowercase letters, especially to avoid confusion. For example, in the case of the electron charge, capital E may be confused with energy. Besides, the literature represents all these constants in lowercase.